### ISSUE TITLE
Wrong results of precision, recall and F1

### ISSUE THREAD

**Issue by del2z** [PyTorchLightning/metrics#439]
I've got wrong results when calculating accuracy, precision, recall and F1.

#### Code sample

```python3
import torch
import torchmetrics.functional as F

preds = torch.tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0])
target = torch.tensor([0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1])

print(F.confusion_matrix(preds, target, num_classes=2))
print(F.precision_recall(preds, target, num_classes=2))
print(F.accuracy(preds, target, num_classes=2))
```
The output is 
```
tensor([[11,  1],
        [ 8,  0]])
(tensor(0.5500), tensor(0.5500))
tensor(0.5500)
```
I got a right confusion matrix, but wrong precision, recall.
The expected values of P, R, F1 are all 0, not the output of 0.5500. 
How should I use these functions in binary case?

### Environment

- PyTorch Version: 1.8.1
- OS: centos 8 (docker env)
- How you installed PyTorch: conda
- Python version: 3.8
- CUDA/cuDNN version: 11.2
- GPU models and configuration: None
- Torchmetrics Version: 0.4.1

**Comment by Borda** [comment]
> The expected values of P, R, F1 are all 0, not the output of 0.5500.

why it shall be 0?

**Comment by del2z** [comment]
My mistake. For binary classification, only P, R, and F1 of positive class are of importance in general case. This is different from multi-class classification. I add `ignore_index=0` in `precision_recall` and got right results.

I suggest using different default values for binary classification would be very convenient for users. Some examples of binary classification are also helpful.
